### PR TITLE
Bug 1128396 - Enable the downloadable dictinary feature

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f4403a64a1e3712cda392427aac1e94ac9686d02

File: Makefile [hunk 1]
```diff
@@ -440,14 +440,17 @@ GAIA_PRETRANSLATE?=1
...
 # including it in the build.
-# Latin IMEngine-backed layouts specified under this variable will have
-# their dictionaries included in the package. The user has to optionally
-# download the dictionary in keyboard settings page in order to enable the
-# layout.
-GAIA_KEYBOARD_PRELOAD_DICT_LAYOUTS?=$(GAIA_KEYBOARD_LAYOUTS)
+# Include the ID of the layout in this variable will make it appear in the
+# keyboard settings page, allowing user to download the dictionary.
+# Any layout listed here that doesn't require a dictionary download and/or
+# comes with it's own dictionary will become a pre-installed layout,
```
Comment (RudyLu) [inline Makefile]: typo: comes with `its`

File: apps/keyboard/build/layout_configurator.js [hunk 2]
```diff
@@ -167,11 +167,11 @@ var KeyboardLayoutConfigurator = function(appDir) {
 };
 
 KeyboardLayoutConfigurator.prototype.loadLayouts =
-function(layoutIds, preloadDictLayoutIds) {
-  var layoutIdSet = this._expandLayoutIdSet(layoutIds);
-  var preloadDictLayoutIdSet = this._expandLayoutIdSet(preloadDictLayoutIds);
+function(enableLayoutIds, downloadableLayoutIds) {
```
Comment (RudyLu) [inline apps/keyboard/build/layout_configurator.js]: enable"d"LayoutIds?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8ce9b8b3824c4d032418eff60bd999d8220af954

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#IJXpf2ZfQV6AyKqOKu4oew

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=952088411aa360d95375443a48b44c580998a7dd

**Pull request merged**